src/cli/commands/force/lightning/local/device/create.ts

```diff
@@ -165,6 +164,27 @@ export class Create extends Setup {
             });
     }
 
+    private _requirements: CommandRequirements = {};
+    public get commandRequirements(): CommandRequirements {
+        if (Object.keys(this._requirements).length === 0) {
+            const requirements: CommandRequirements = {};
+            requirements.setup = getPlatformSetupRequirements(
+                this.logger,
+                this.flags.platform,
+                this.flags.apilevel
+            );
+            requirements.create = {
+                requirements: [
+                    new DeviceNameAvailableRequirement(this, this.logger),
+                    new ValidDeviceTypeRequirement(this, this.logger)
+                ],
+                enabled: true
```

Comment:
Preview is a bit different because we need to wait until the runtime to get the flag value. That said, the next round of refactor work shall address this issue and provide more consistent approach. If you want I can still do the same as preview but the primary goal for this PR was to get it functional with the latest `core`. Let me know!